### PR TITLE
Clarifies installation/runtime options for CI/PROD images.

### PR DESCRIPTION
**Pull request by potiuk**
After PROD images were added, some of the flags had two meanings

These behaved differently in PROD image and CI image and were the
source of confusion especially when start-airflow command was used.

For PROD image, the image can be customized during image building,
and packages could be installed from .whl or .sdist packages
available in `docker-context-files`. This
is used at CI and dockerhub building time to produce image built
packages that were prepared using local sources.

The CI image is always built from local sources but airflow can
be removed and re-installed at runtime from pypi.
Both airflow and provider packages can be installed
from .whl or .sdist packages available in dist folder. This is
used in CI to test current provider packages with older
Airflow released (2.0.0) and to test provider packages locally.

After the change we have two sets of flags/variables:

PROD image (building image):

* install-airflow-version, install-airflow-reference,
  install-from-docker-context-files

CI image (runtime):

* use-airflow-version, use-packages-from-dist

That should avoid confusion and failures of commands such as
`start-airflow` that is used to test provider packages and
airflow itself.

---
**^ Add meaningful description above**

Read the **[Pull Request Guidelines](https://github.com/apache/airflow/blob/master/CONTRIBUTING.rst#pull-request-guidelines)** for more information.
In case of fundamental code change, Airflow Improvement Proposal ([AIP](https://cwiki.apache.org/confluence/display/AIRFLOW/Airflow+Improvements+Proposals)) is needed.
In case of a new dependency, check compliance with the [ASF 3rd Party License Policy](https://www.apache.org/legal/resolved.html#category-x).
In case of backwards incompatible changes please leave a note in [UPDATING.md](https://github.com/apache/airflow/blob/master/UPDATING.md).

### CODE REVIEW

**Comment by potiuk** [comment]
This solves problem https://apache-airflow.slack.com/archives/CQ9QHSFQX/p1617988430029300  - which also other people reported on slack

**Comment by potiuk** [comment]
would be nice to fix it

**Comment by uranusjr** [comment]
I’m most just curious here, what happens if you pass `--install-airflow-version` on CI, and vice versa? No effect?

**Comment by potiuk** [comment]
> I’m most just curious here, what happens if you pass `--install-airflow-version` on CI, and vice versa? No effect?

After this change - yes. No effect.

**Comment by uranusjr** [comment]
Awesome, thanks.

File: scripts/ci/libraries/_initialization.sh [hunk 1]
```diff
@@ -464,6 +464,13 @@ function initialization::initialize_image_build_variables() {
     #   * 'constraints-source-providers' for constraints with source version of providers (defaults in Breeze and CI)
     #   * 'constraints-no-providers' for constraints without providers
     export AIRFLOW_CONSTRAINTS="${AIRFLOW_CONSTRAINTS:="constraints-source-providers"}"
+
+    # Replace airflow at runtime in CI image with the one specified
+    #   * none - just removes airflow
+    #   * wheel - replaces airflow with one specified in the wheel file in /dist
+    #   * wheel - replaces airflow with one specified in the sdist file in /dist
```
Comment (kaxil) [inline scripts/ci/libraries/_initialization.sh:471]:
```suggestion
    #   * sdist - replaces airflow with one specified in the sdist file in /dist
```

**Comment by kaxil** [comment]
For some reason the tests timed out, can you rebase to master please

**Comment by potiuk** [comment]
Green :)

**Review by kaxil — APPROVED** [review]

**Pull request merged**